src/commands/commitCommands.ts

```diff
@@ -38,6 +38,18 @@ export async function commit({ repository, switches }: MenuState, commitArgs: st
   return runCommitLikeCommand(repository, args);
 }
 
+export async function fixup({ repository, switches }: MenuState) {
+  const sha = await MagitUtils.chooseCommit(repository, 'Fixup commit');
+
+  if (sha) {
+    const args = ['commit', ...MenuUtil.switchesToArgs(switches), '--fixup', sha];
+
+    return runCommitLikeCommand(repository, args);
```

Comment:
Will a `fixup` never invoke the editor?
If so runCommitLikeCommand is unnecessary and a bit jarring as it presents the staged changes and closes them immediately.
If it doesnt use the editor, this will suffice
```suggestion
    return await gitRun(repository, args);
```